src/store/safeOverviews.ts

```diff
@@ -0,0 +1,164 @@
+import { createApi } from '@reduxjs/toolkit/query/react'
+
+import { type SafeOverview, getSafeOverviews } from '@safe-global/safe-gateway-typescript-sdk'
+import { sameAddress } from '@/utils/addresses'
+import type { RootState } from '.'
+import { selectCurrency } from './settingsSlice'
+import { type SafeItem } from '@/components/welcome/MyAccounts/useAllSafes'
+
+const noopBaseQuery = async () => ({ data: null })
+
+type SafeOverviewQueueItem = {
+  safeAddress: string
+  walletAddress?: string
+  chainId: string
+  currency: string
+  callback: (result: { data: SafeOverview | undefined; error?: never } | { data?: never; error: string }) => void
+}
+
+const BATCH_SIZE = 5
+const FETCH_TIMEOUT = 200
+
+const makeSafeId = (chainId: string, address: string) => `${chainId}:${address}` as `${number}:0x${string}`
+
+class SafeOverviewFetcher {
+  private requestQueue: SafeOverviewQueueItem[] = []
+
+  private fetchTimeout: NodeJS.Timeout | null = null
+
+  private async fetchSafeOverviews({
+    safeIds,
+    walletAddress,
+    currency,
+  }: {
+    safeIds: `${number}:0x${string}`[]
+    walletAddress?: string
+    currency: string
+  }) {
+    return await getSafeOverviews(safeIds, {
+      trusted: true,
+      exclude_spam: true,
+      currency,
+      wallet_address: walletAddress,
+    })
+  }
+
+  private async processQueuedItems() {
+    // Dequeue the first BATCH_SIZE items
+    const nextBatch = this.requestQueue.slice(0, BATCH_SIZE)
+    this.requestQueue = this.requestQueue.slice(BATCH_SIZE)
+
+    console.log('[SafeOverviewFetcher] processing Queued Items', nextBatch, [...this.requestQueue])
+
+    let overviews: SafeOverview[]
+    try {
+      this.fetchTimeout && clearTimeout(this.fetchTimeout)
+      this.fetchTimeout = null
+
+      const safeIds = nextBatch.map((request) => makeSafeId(request.chainId, request.safeAddress))
+      const { walletAddress, currency } = nextBatch[0]
+      overviews = await this.fetchSafeOverviews({ safeIds, currency, walletAddress })
+    } catch (err) {
+      // Overviews could not be fetched
+      nextBatch.forEach((item) => item.callback({ error: 'Could not fetch Safe overview' }))
+      return
+    }
+
+    nextBatch.forEach((item) => {
+      const overview = overviews.find(
+        (entry) => sameAddress(entry.address.value, item.safeAddress) && entry.chainId === item.chainId,
+      )
+
+      item.callback({ data: overview })
+    })
```

Comment:
since we're already calling the item.callback in the catch block, shouldn't this line be inside the try block?